feat: default to fed2 when @link is present (#1097)
apollographql/apollo-cli: docs/source/supergraphs.md

**Apollo Gateway fails to start up if it's provided with a supergraph schema that it doesn't support.** To ensure compatibility, we recommend that you test launching your gateway in a CI pipeline with the supergraph schema it will ultimately use in production.

#### Federation 1 vs. Federation 2

The `rover supergraph compose` command generates a supergraph schema via [composition](https://www.apollographql.com/docs/federation/federated-types/composition/). For Federation 1, this algorithm was implemented in the [`@apollo/federation`](https://www.npmjs.com/package/@apollo/federation) package. For Federation 2, this algorithm is implemented in the [`@apollo/composition`](https://www.npmjs.com/package/@apollo/composition) package.

> `rover supergraph compose` supports composing subgraphs with both Federation 1 and Federation 2. By default, Rover will use Federation 2 if and only if one of your subgraphs contains an `@link` directive, otherwise it will use Federation 1. When Apollo releases new versions of composition for Federation 1 or Federation 2, Rover downloads the new package to your machine. Rover then uses the new composition package to compose your subgraphs. Our aim is to release only backward-compatible changes across major versions moving forward. If composition breaks from one version to the next, please [submit an issue](https://github.com/apollographql/federation/issues/new?assignees=&labels=&template=bug.md), and follow the instructions for pinning composition to a known good version.

Reviewer: ```suggestion
`rover supergraph compose` supports composing subgraphs with both Federation 1 and Federation 2. By default, Rover uses Federation 2 composition _if and only if_ at least one of your subgraph schemas contains the `@link` directive. Otherwise, it uses  Federation 1.

When Apollo releases new versions of composition for Federation 1 or Federation 2, Rover downloads the new package to your machine. Rover then uses the new composition package to compose your subgraphs. Our aim is to release only backward-compatible changes across major versions moving forward. If composition breaks from one version to the next, please [submit an issue](https://github.com/apollographql/federation/issues/new?assignees=&labels=&template=bug.md), and follow the instructions for pinning composition to a known good version.
```

If you set `federation_version: 1` or `federation_version: 2`, you can run `rover supergraph compose` with the `--skip-update` flag to prevent Rover from downloading newer composition versions. Rover instead uses the latest major version that you've downloaded to your machine. This can be helpful if you're on a slow network.

> If a subgraph contains an `@link` directive, and you've specified `federation_version: 1`, you will need to either downgrade your subgraph, or upgrade to Federation 2.

Reviewer: ```suggestion
If any subgraph schema contains the `@link` directive and you've specified `federation_version: 1`, you either need to revert that subgraph to a Federation 1 schema or move your graph to Federation 2.
```
